include/aspect/particle/particle_handler.h

```diff
@@ -171,13 +277,33 @@ namespace aspect
         const std::multimap<types::LevelInd, Particle<dim,spacedim> > &
         get_particles() const;
 
+        /**
+         * Const access to particles in this world.
+         * TODO: This function needs to go to not expose internal structure.
+         * This can only be done once World no longer uses this function.
```

Comment:
Are you worried about exposing this function because it shows the underlying data structure when you carefully hide the data structure for the locally owned particles?